packages/rest/src/providers/invoke-method.provider.ts

```diff
@@ -19,3 +20,23 @@ export class InvokeMethodProvider implements Provider<InvokeMethod> {
     return route.invokeHandler(this.context, args);
   }
 }
+
+@bind(
+  asMiddleware({group: 'invokeMethod', chain: RestTags.REST_MIDDLEWARE_CHAIN}),
+)
+export class InvokeMethodMiddlewareProvider implements Provider<Middleware> {
+  constructor(
+    @inject(RestBindings.SequenceActions.INVOKE_METHOD)
+    protected invokeMethod: InvokeMethod,
+  ) {}
+
+  value(): Middleware {
+    return async (ctx, next) => {
+      const route: RouteEntry = await ctx.get(RestBindings.Operation.ROUTE);
+      const params: OperationArgs = await ctx.get(
+        RestBindings.Operation.PARAMS,
+      );
+      return this.invokeMethod(route, params);
```

Comment:
why is `next` not called here ? for this to be an independent middleware function, it should have no knowledge that it is the last in the middleware chain. The invocation part should handle that.

Reply:
It's the middleware logic's decision not to call `next` as `invokeMethod` returns the result.

Reply:
ok I see; so next() is just a customary function for the middleware, it does not really link with the middleware function in the chain ?

Reply:
that is why the sequence handle is forced to invoke the middleware functions one by one 
 ``` 
for (const opt of this.options) {
      // Invoke registered Express middleware
      debug(
        'Invoking middleware chain %s with groups %s',
        opt.chain,
        opt.orderedGroups,
      );
      const done = await this.invokeMiddleware(context, opt);
      if (done) break;
    } 
```

Reply:
> ok I see; so next() is just a customary function for the middleware, it does not really link with the next middleware function in the chain ?

It does. But it's up the middleware logic to decide if the next one should be invoked. The `invokeMethod` is designed to invoke the controller method directly. 

> why sequence.handle() is forced to invoke the middleware functions one by one?
This is to allow multiple chains to be used one by one. If you need to cascade them, add the chain as a middleware or flatten the chains.